DOC: added note that answer tests are currently run with nose (#4955)
yt-project/yt: doc/source/developing/testing.rst

Answer Testing
--------------
.. note::
    This section documents answer tests run with ``pytest``. The plan is to
    switch to using ``pytest`` for answer tests at some point in the future,

>     but currently (July 2024), answer tests are actually run using ``nose``.

Reviewer: ```suggestion
    but currently (July 2024), answer tests are still implemented and run with ``nose``.
```

>     but currently (July 2024), answer tests are still implemented and run with ``nose``.

Reviewer 2: also a question for @neutrinoceros but would it be helpful to link to an old version of the docs where the nose testing is described. e.g., https://yt-project.org/docs/4.0.0/developing/testing.html#answer-testing ? I'd rather not encourage folks to write new tests for nose if they can avoid it, but there may be situations where unfortunately it is required (e.g., writing a new frontend).

Reviewer: that's a great idea. I didn't know that old docs were still accessible !

Reviewer 2: yup! and I have on occasion gone back to look at the nose testing docs when i find myself needing to run nosetests locally. So in that case, @nastasha-w , mind adding the following (or something similar)?

```suggestion
    but currently (July 2024), answer tests are still implemented and run with ``nose``.
    If needed, you can refer to the old documentation on answer testing `here <https://yt-project.org/docs/4.0.0/developing/testing.html#answer-testing>`_.
```

Author: seems reasonable. That is basically the situation in PR #4939, where changes to the projection backend are causing image comparison tests of the frontends to fail

Author: I've added some extra text with the link. Sorry it took a while; I forgot to set the environment variable for a quick docs build
